app/components/newsletter/newsletter-header-overview.js

```diff
@@ -106,73 +72,151 @@ export default class NewsletterHeaderOverviewComponent extends Component {
 
     await reloadedMeeting.save();
     this.isLoading = false;
-    location.reload();
   }
 
   @action
-  async sendCampaign() {
+  async publishToMail() {
     this.isLoading = true;
-    const agenda = this.args.agenda;
-    const meeting = this.args.meeting;
-    const mailCampaign = await meeting.mailCampaign;
+    const mailCampaign = await this.getMailCampaign();
 
-    if (!mailCampaign || !mailCampaign.id || mailCampaign.isSent) {
-      return;
+    if (mailCampaign) {
+      if (await this.validateMailCampaign(mailCampaign)) {
+        await this.publishToMailAndSaveCampaign(mailCampaign);
+      }
+    }
+
+    this.isLoading = false;
+    this.toggleVerifyingPublishMail();
+  }
+
+  @action
+  async publishToBelga() {
+    this.isLoading = true;
+    await this.publishNewsletterToBelga();
+    this.isLoading = false;
+    this.toggleVerifyingPublishBelga();
+  }
+
+  @action
+  async publishToAll() {
+    this.isLoading = true;
+
+    const mailCampaign = await this.getMailCampaign();
+    if (mailCampaign) {
+      if (await this.validateMailCampaign(mailCampaign)) {
+        await this.publishToMailAndSaveCampaign(mailCampaign);
+      }
+    }
```

Comment:
code duplicatie, dit doe je ook in `publishToMail`.
Ook is het veel te vaag wat er hier juist gebeurt.
In de `getMailCampaign` return je ofwel null (al verzonden) + error popup, ofwel een nieuw aangemaakte campaign, ofwel een bestaande campaign.
Daarna ga je die campaign valideren, maar in geval van die nieuwe is het net aangemaakt en zou niet gevalideerd moeten worden?
Terzijde: Het valideren van de mailCampagne zou ook kunnen gebeuren in de `newsletter-service` zelf, voor we gaan versturen.